src/front/glsl/builtins.rs

```diff
@@ -445,12 +445,19 @@ pub fn inject_builtin(declaration: &mut FunctionDeclaration, module: &mut Module
             // bits layout
             // bit 0 - dim part 1 - 1D/2D
             // bit 1 - array
-            // bit 2 - dim part 2 - 3D
+            // bit 2 - multisample
+            // bit 3 - dim part 2 - 3D
```

Comment:
actually, why aren't we doing `bit 0-1 - dimension 1D/2D/3D` ?

Reply:
Because you have to iterate less times and have less checks, but honestly this system doesn't work out as well as I initially thought and I'm tempted to just replace it with nested loops with proper types and free standing functions to reduce the right drift.

Reply:
I don't think you need to iterate any less. In either case there are 2 bits for the dimension. I'm just saying let's put them together and treat as a number between 0 and 4 instead of 2 separate bits.